pandas/core/tools/datetimes.py

```diff
@@ -539,19 +551,70 @@ def _adjust_to_origin(arg, origin, unit):
     return arg
 
 
+@overload
 def to_datetime(
-    arg,
-    errors="raise",
-    dayfirst=False,
-    yearfirst=False,
-    utc=None,
-    format=None,
-    exact=True,
-    unit=None,
-    infer_datetime_format=False,
+    arg: DatetimeScalar,
+    errors: str = ...,
+    dayfirst: bool = ...,
+    yearfirst: bool = ...,
+    utc: Optional[bool] = ...,
+    format: Optional[str] = ...,
+    exact: bool = ...,
+    unit: Optional[str] = ...,
+    infer_datetime_format: bool = ...,
+    origin=...,
+    cache: bool = ...,
+) -> Timestamp:
```

Comment:
or NaT?

Reply:
The return type here is dependent of the **value** of the errors parameter. so to refine the overload further requires Literal. 

updated to a Union return type with NaT but needs a cast (or ignore/assert) in  pandas/io/excel/_odfreader.py

also changed from Timestamp to DatetimeScalar since with error="ignore" returns arg value

Reply:
> so to refine the overload further requires Literal

FWIW I think (separate to this PR) that we will need to start vendoring parts of typing_extensions and this would be a good candidate. I also think Protocol is necessary for us to unpin where we are at now particularly with our mixin classes

Reply:
+1 for vendoring typing_extensions